var/spack/repos/builtin/packages/relion/package.py

```diff
@@ -65,14 +92,20 @@ def cmake_args(self):
             '-DDoublePrec_GPU=%s' % ('+double-gpu' in self.spec),
         ]
 
-        carch = self.spec.variants['cuda_arch'].value
-
         if '+cuda' in self.spec:
-            args += [
-                '-DCUDA=on',
-            ]
-            if carch is not None:
-                args += [
-                    '-DCUDA_ARCH=%s' % (carch),
-                ]
+            # relion+cuda requires selecting cuda_arch
+	    if not carch: 
+		sys.stdout.write('relion+cuda requires selecting cuda_arch')
```

Comment:
This should be a `conflicts` directive.

Reply:
Sorry, @adamjstewart I have no idea how to do this

Can conflicts() go under the cmake_args function?  

I tried:
```
    if '+cuda' in self.spec:
            # relion+cuda requires selecting cuda_arch
            if not carch:
                conflicts('%gcc')
```
For example which does not stop execution.

Reply:
Nope, it has to be a top-level directive.

Reply:
How do I reference spec.variants['cuda_arch'].value?
```
      test = spec.variants['cuda_arch'].value
      print(test)
      sys.exit()
  ```

  File "/home/user/prepos/spack/var/spack/repos/builtin/packages/relion/package.py", line 30, in <module>
    """RELION (for REgularised LIkelihood OptimisatioN, pronounce rely-on) is a
  File "/home/dod2014/prepos/spack/var/spack/repos/builtin/packages/relion/package.py", line 81, in Relion
    print(test)
NameError: name 'spec' is not defined

Reply:
Think I have an idea.  Always `grep` before asking questions.

Reply:
Never mind.

How do I use `spec.variants` outside the cmake function?

Reply:
`spack.variant.spack.spec` 

I need to start using Eclipse or make vim a little more useful.